src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -115,14 +114,15 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
             Type = type;
             Options = options;
 
-            ClassType = GetClassType(
-                type,
-                parentClassType: type,
-                propertyInfo: null,
-                out Type? runtimeType,
-                out Type? elementType,
-                out JsonConverter? converter,
-                options);
+            JsonConverter converter = GetConverter(
+                Type,
+                parentClassType: null, // A ClassInfo never has a "parent" class.
+                propertyInfo: null, // A ClassInfo never has a "parent" property.
+                out Type runtimeType,
+                Options);
+
+            ClassType = converter.ClassType;
+            PolicyProperty = CreatePolicyProperty(Type, runtimeType, converter, Options);
```

Comment:
I did this previously in the last iteration and based on our discussion I removed it since it affected so many files for this PR. It should be its own PR. Currently there is no issue although I can create that PR right after this.